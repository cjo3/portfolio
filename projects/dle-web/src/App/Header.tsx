import { switchLinkRoutePath, createHashId } from '@cjo3/shared/react/helpers'
import { Button, Grid, Typography } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'
import React from 'react'

const useStyles = makeStyles(
  theme => ({
    Header_containerBg: {
      backgroundColor: ({ bgColor }) => eval(bgColor),
      backgroundImage: ({ bgUrls }) => `url(${bgUrls[0]})`,
      backgroundPosition: 'center',
      backgroundSize: 'cover',
      backgroundRepeat: 'no-repeat',
      [theme.breakpoints.up('sm')]: {
        backgroundImage: ({ bgUrls }) => `url(${bgUrls[1]})`
      },
      [theme.breakpoints.up('md')]: {
        backgroundImage: ({ bgUrls }) => `url(${bgUrls[2]})`
      },
      [theme.breakpoints.up('lg')]: {
        backgroundImage: ({ bgUrls }) => `url(${bgUrls[3]})`
      },
      [theme.breakpoints.up('xl')]: {
        backgroundImage: ({ bgUrls }) => `url(${bgUrls[4]})`
      }
    },
    Header_contentContainer: {
      ...theme.custom.contentContainer,
      ...theme.custom.setFlex('column'),
      height: 300,
      padding: theme.custom.setSpace('sm')
    },
    Header_title: {
      color: ({ titleColor }) => eval(titleColor),
      marginTop: theme.custom.setSpace(),
      textShadow: theme.custom.textShadow
    },
    Header_subTitle: {
      color: ({ subTitleColor }) => eval(subTitleColor),
      marginTop: theme.custom.setSpace(),
      textShadow: theme.custom.textShadow,
      fontWeight: 'normal'
    },
    Header_button: {
      marginTop: theme.custom.setSpace('sm')
    }
  }),
  {
    name: `Header-${createHashId()}`
  }
)

interface Props {
  title: string
  titleColor?: string
  subTitle?: string
  subTitleColor?: string
  bgColor?: string
  bgUrls?: string[]
  buttonHref?: string
  buttonLabel?: string
}

export const Header: React.FC<Props> = ({
  title,
  titleColor = 'theme.palette.grey[50]',
  subTitle,
  subTitleColor = 'theme.palette.grey[300]',
  bgColor,
  bgUrls,
  buttonHref,
  buttonLabel
}): JSX.Element => {
  const classes = useStyles({
    titleColor,
    subTitleColor,
    bgColor,
    bgUrls
  })

  return (
    <Grid
      component="header"
      container
      justify="center"
      alignItems="center"
      className={classes.Header_containerBg}>
      <Grid className={classes.Header_contentContainer}>
        <Typography
          variant="h1"
          align="center"
          className={classes.Header_title}>
          {title}
        </Typography>
        {subTitle && (
          <Typography
            variant="h4"
            align="center"
            className={classes.Header_subTitle}>
            {subTitle}
          </Typography>
        )}
        {buttonHref && (
          <Button
            type="button"
            color="secondary"
            variant="contained"
            className={classes.Header_button}
            href={switchLinkRoutePath(buttonHref)}>
            {buttonLabel}
          </Button>
        )}
      </Grid>
    </Grid>
  )
}
