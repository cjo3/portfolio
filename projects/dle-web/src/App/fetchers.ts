import { addAuthHeaderToOptions } from '@cjo3/shared/security/authToken'
import post from 'axios'

const otherHeaders = {
  headers: {
    accept: 'application/json'
  }
}

export const makePreOrder = async (email: string) => {
  const apiUrl =
    process.env.NODE_ENV === 'production'
      ? `${process.env.API_URL}/${process.env.CDN_APP_FOLDER}`
      : 'http://localhost:2000'

  const res = await post(
    `${apiUrl}/payment`,
    { email },
    addAuthHeaderToOptions(otherHeaders)
  )
  return res.data
}
