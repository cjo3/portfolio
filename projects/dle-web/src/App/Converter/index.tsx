import { BackDropScreen } from '@cjo3/shared/react/components/BackDropScreen'
import Loadable from 'react-loadable'
import { Grid } from '@material-ui/core'
import React from 'react'
import { ConverterBg } from '../../assets'
import { Header } from '../Header'

const ConverterAppLoadable = Loadable({
  loader: () =>
    import(
      /* webpackChunkName: "chunk-ConverterApp" */
      /* webpackPrefetch: false */
      './ConverterApp'
    ),
  loading: ({ error, pastDelay, timedOut }) => {
    if (timedOut) return <h1>Timed Out</h1>
    if (error) return <h1>Failed to Load</h1>
    if (pastDelay) return <BackDropScreen isOpen spinner />
    return null
  },
  delay: 250,
  timeout: 5000,
  render: (loaded, props) => {
    const Component = loaded.ConverterApp
    return <Component {...props} />
  }
})

export const Converter: React.FC = (): JSX.Element => {
  return (
    <Grid container justify="center">
      <Header
        title="Converter"
        subTitle="Load a File and Process your Sheet"
        bgColor="theme.palette.primary.main"
        bgUrls={ConverterBg.paths}
        buttonHref="/converter/guide"
        buttonLabel="Read the Guide"
      />
      <ConverterAppLoadable />
    </Grid>
  )
}
