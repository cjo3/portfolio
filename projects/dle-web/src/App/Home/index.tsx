import { Grid, Typography } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'
import clsx from 'clsx'
import { ImageHandler } from '@cjo3/shared/react/components/ImageHandler'
import React from 'react'
import {
  HomeHeader,
  HomeEnvironment,
  HomeStarBg,
  HomeGeochemistry,
  HomeOilGas,
  HomeFoodSafety,
  HomePaperPulp,
  HomePharmaceutical
} from '../../assets'
import { Header } from '../Header'

const useStyles = makeStyles(theme => ({
  Home_contentContainer: {
    ...theme.custom.contentContainer
  },
  Home_introContainerBg: {
    background: theme.custom.setLinearGradient(
      180,
      theme.palette.grey[300],
      'white'
    )
  },
  Home_introContent: {
    padding: `${theme.custom.setSpace('md')}px ${theme.custom.setSpace('sm')}px`
  },
  Home_introImage: {
    marginBottom: theme.custom.setSpace('sm')
  },
  Home_useCaseContainerBg: {
    background: theme.custom.setLinearGradient(
      180,
      theme.palette.grey[400],
      theme.palette.grey[300]
    )
  },
  Home_useCaseContent: {
    padding: `${theme.custom.setSpace('md')}px ${theme.custom.setSpace('sm')}px`
  },
  Home_useCaseContainer: {
    marginTop: theme.custom.setSpace('sm'),
    ...theme.custom.setGrid(3, 'auto', theme.custom.setSpace('sm')),
    width: '100%'
  },
  Home_useCase0: {
    gridColumn: '1 / 2',
    gridRow: 1,
    backgroundImage: `url(${HomeEnvironment.paths[0]})`,
    [theme.breakpoints.down('xs')]: {
      gridColumn: '1 / 4'
    }
  },
  Home_useCase1: {
    gridColumn: '2 / 3',
    gridRow: 1,
    backgroundImage: `url(${HomeGeochemistry.paths[0]})`,
    [theme.breakpoints.down('xs')]: {
      gridColumn: '1 / 4',
      gridRow: 2
    }
  },
  Home_useCase2: {
    gridColumn: '3 / 4',
    gridRow: 1,
    backgroundImage: `url(${HomeOilGas.paths[0]})`,
    backgroundPosition: 'top middle',
    [theme.breakpoints.down('xs')]: {
      gridColumn: '1 / 4',
      gridRow: 3
    }
  },
  Home_useCase3: {
    gridColumn: '1 / 2',
    gridRow: 2,
    backgroundImage: `url(${HomePharmaceutical.paths[0]})`,
    [theme.breakpoints.down('xs')]: {
      gridColumn: '1 / 4',
      gridRow: 4
    }
  },
  Home_useCase4: {
    gridColumn: '2 / 3',
    gridRow: 2,
    backgroundImage: `url(${HomePaperPulp.paths[0]})`,
    [theme.breakpoints.down('xs')]: {
      gridColumn: '1 / 4',
      gridRow: 5
    }
  },
  Home_useCase5: {
    gridColumn: '3 / 4',
    gridRow: 2,
    backgroundImage: `url(${HomeFoodSafety.paths[0]})`,
    [theme.breakpoints.down('xs')]: {
      gridColumn: '1 / 4',
      gridRow: 6
    }
  },
  Home_useCase: {
    'padding': theme.custom.setSpace('sm'),
    ...theme.custom.setFlex(),
    ...theme.custom.borderRadius,
    'boxShadow': theme.custom.boxShadow,
    'color': theme.palette.grey[50],
    'textShadow': theme.custom.textShadow,
    'backgroundSize': 'cover',
    'backgroundRepeat': 'no-repeat',
    'backgroundBlendMode': 'multiply',
    'backgroundColor': 'rgba(0, 0, 0, 0.5)',
    'transition': 'background-color 250ms ease-out',
    '&:hover': {
      backgroundColor: 'rgba(0, 0, 0, 0.25)'
    }
  }
}))

export const Home: React.FC = (): JSX.Element => {
  const classes = useStyles()

  return (
    <Grid container>
      <Header
        name="Home"
        title={process.env.APP_NAME}
        subTitle="Conversion tool for spreadsheet-based, raw analytical data"
        bgColor="theme.palette.primary.main"
        bgUrl={HomeStarBg.paths[0]}
        buttonHref="/converter"
        buttonLabel="Start"
      />
      <Grid
        component="section"
        container
        justify="center"
        className={classes.Home_introContainerBg}>
        <Grid
          container
          justify="center"
          direction="column"
          alignItems="center"
          className={clsx(
            classes.Home_contentContainer,
            classes.Home_introContent
          )}>
          <ImageHandler
            asset={HomeHeader}
            outerClass={classes.Home_introImage}
          />
          <Typography variant="h3" align="center">
            Easily convert MDL text to actionable number values
          </Typography>
          <Typography variant="body1" align="center">
            This app is suited for individuals who import analytical data into
            proprietary software that does not already have such editing
            features built-in. {process.env.APP_NAME} bypasses the tedious
            operation of &quot;Find &amp; Replace&quot; used in Excel and other
            spreadsheet programs. Now, you can replace those limiting threshold
            characters in one step by just selecting your parameters and
            applying them to your entire data range. The revised data can be
            reviewed prior to downloading; export formats are currently just
            Excel &#40;.xls and .xlsx&#41;. first pass and some text for home
            page summary
          </Typography>
        </Grid>
      </Grid>

      <Grid
        component="section"
        container
        justify="center"
        className={classes.Home_useCaseContainerBg}>
        <Grid
          container
          justify="center"
          className={clsx(
            classes.Home_contentContainer,
            classes.Home_useCaseContent
          )}>
          <Typography variant="h3" align="center">
            Applied Science Applications
          </Typography>
          <div className={classes.Home_useCaseContainer}>
            <div className={clsx(classes.Home_useCase0, classes.Home_useCase)}>
              <Typography variant="h5" align="center">
                Environmental Testing
              </Typography>
            </div>
            <div className={clsx(classes.Home_useCase1, classes.Home_useCase)}>
              <Typography variant="h5" align="center">
                Geochemistry
              </Typography>
            </div>
            <div className={clsx(classes.Home_useCase2, classes.Home_useCase)}>
              <Typography variant="h5" align="center">
                Oil &amp; Gas
              </Typography>
            </div>
            <div className={clsx(classes.Home_useCase3, classes.Home_useCase)}>
              <Typography variant="h5" align="center">
                Pharmaceutical
              </Typography>
            </div>
            <div className={clsx(classes.Home_useCase4, classes.Home_useCase)}>
              <Typography variant="h5" align="center">
                Pulp &amp; paper
              </Typography>
            </div>
            <div className={clsx(classes.Home_useCase5, classes.Home_useCase)}>
              <Typography variant="h5" align="center">
                Food Safety
              </Typography>
            </div>
          </div>
        </Grid>
      </Grid>
    </Grid>
  )
}
