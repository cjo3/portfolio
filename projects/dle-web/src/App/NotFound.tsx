import { Header } from './Header'
import { Grid } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'
import React from 'react'
import { NotFoundBg } from '../assets'

const useStyles = makeStyles(
  theme => ({
    NotFound_container: {
      height: '100%'
    }
  }),
  {
    name: 'NotFound'
  }
)

export const NotFound: React.FC = (): JSX.Element => {
  const classes = useStyles()
  return (
    <Grid container className={classes.NotFound_container}>
      <Header
        title="Error"
        subTitle="Sorry, something went wrong"
        bgColor="theme.palette.grey[900]"
        bgUrl={NotFoundBg.paths[0]}
        buttonHref="/"
        buttonLabel="Home"
      />
    </Grid>
  )
}
