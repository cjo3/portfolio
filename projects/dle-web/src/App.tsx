import React from 'react'
import { Route, Switch } from 'react-router-dom'

import { Home } from './Home'
import { NotFound } from './NotFound'

export const App: React.FC = () => (
  <Switch>
    <Route path="/" exact component={() => <Home name="Dave" size={23} />} />
    <Route path="/*" component={NotFound} />
  </Switch>
)
