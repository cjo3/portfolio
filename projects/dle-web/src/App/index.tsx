import { setTracker, switchLinkRoutePath } from '@cjo3/shared/react/helpers'
import { CssBaseline, Snackbar } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'
import MuiAlert from '@material-ui/lab/Alert'
import React, { useEffect, useLayoutEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { useLocation } from 'react-router'
import { Route, Switch } from 'react-router-dom'
import { SNACKBAR_TIMEOUT } from '../constants'
import {
  addTracker,
  closeSnackbar,
  addModernizrFeature
} from '../store/app/actions'
import { Snackbar as ISnackbar } from '../store/app/interfaces'
import {
  snackbarSelector,
  trackerSelector,
  webpOkSelector
} from '../store/selectors'
import { Converter } from './Converter'
import { Footer } from './Footer'
import { Home } from './Home'
import { NotFound } from './NotFound'
import { TopNav } from './TopNav'
import { Guide } from './Converter/Guide/'

const useStyles = makeStyles(
  theme => ({
    App_pageContainer: {
      ...theme.custom.setGrid(12, 'auto 1fr auto'),
      ...theme.custom.fullScreen
    },
    App_topNavPosition: {
      gridColumn: '1 / 13',
      gridRow: 1
    },
    App_contentPosition: {
      ...theme.custom.setFlex('column', 'flex-start'),
      gridColumn: '1 / 13',
      gridRow: 2
    },
    App_footerPosition: {
      gridColumn: '1 / 13',
      gridRow: 3
    }
  }),
  {
    name: 'App'
  }
)

const checkWebpOk = async dispatchHandler => {
  import(
    /* webpackChunkName: "chunk-modernizr" */
    'modernizr'
  ).then(({ default: mod }) => {
    mod.on('webp', status => {
      if (status.lossless && status.alpha) {
        dispatchHandler(addModernizrFeature('webp'))
      }
    })
  })
}

export const App: React.FC = (): JSX.Element => {
  const classes = useStyles()

  const dispatch = useDispatch()

  let withNav = false

  const location = useLocation()

  withNav = /\/converter\/?(guide\/?)?$/.test(location.pathname)

  const webpOk = useSelector(webpOkSelector)

  if (!process.env.IS_SERVER) {
    useLayoutEffect(() => {
      if (!webpOk) checkWebpOk(dispatch)
    }, [])
  }

  const snackbar: ISnackbar = useSelector(snackbarSelector)

  let snackbarTimeout: any

  const closeSnackbarHandler = (
    event: React.MouseEvent<HTMLButtonElement>,
    reason: string
  ): void => {
    if (snackbarTimeout && reason === 'clickaway') {
      dispatch(closeSnackbar())
      clearTimeout(snackbarTimeout)
    } else {
      snackbarTimeout = setTimeout(() => {
        dispatch(closeSnackbar())
      }, SNACKBAR_TIMEOUT)
    }
  }

  if (!process.env.IS_SERVER) {
    useLayoutEffect(() => {
      const styleTags = document.getElementsByTagName('style')
      const injectionPoint = document.getElementById('jssInjectionPoint')

      if (styleTags.length > 0) {
        for (let tag of styleTags) {
          if (tag.dataset?.meta && tag.dataset.meta === 'MuiSvgIcon') {
            injectionPoint.insertAdjacentElement('beforebegin', tag)
          }
        }
      }
    })
  }

  let tracker = useSelector(trackerSelector)

  useEffect(() => {
    if (!tracker) {
      tracker = setTracker(process.env.GA_TAG)
      tracker.initialize()
      dispatch(addTracker(tracker))
    }
  }, [tracker])

  useEffect(() => {
    tracker.pageHit(process.env.APP_ROOT_PATH, location.pathname)
  }, [location])

  return (
    <CssBaseline>
      <div className={classes.App_pageContainer}>
        <Snackbar
          key={snackbar.key}
          anchorOrigin={{
            vertical: 'top',
            horizontal: 'left'
          }}
          open={snackbar.open}
          onClose={closeSnackbarHandler}>
          <MuiAlert elevation={6} variant="filled" severity={snackbar.severity}>
            {snackbar.message}
          </MuiAlert>
        </Snackbar>
        {withNav && <TopNav style={classes.App_topNavPosition} />}
        <div className={classes.App_contentPosition}>
          <Switch>
            <Route path={switchLinkRoutePath('/')} exact component={Home} />
            <Route
              path={switchLinkRoutePath('/converter')}
              exact
              component={Converter}
            />
            <Route
              path={switchLinkRoutePath('/converter/guide')}
              exact
              component={Guide}
            />
            <Route path="/*" component={NotFound} />
          </Switch>
        </div>
        <Footer style={classes.App_footerPosition} />
      </div>
    </CssBaseline>
  )
}
