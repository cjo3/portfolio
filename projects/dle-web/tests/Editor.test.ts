import { testRenderer, renderSnapshot } from './setup'
import { Editor } from '../src/App/Editor'

describe('Editor', () => {
  describe('render', () => {
    test('matches current snapshop', () => {
      const tree = renderSnapshot('/editor', Editor)
      expect(tree).toMatchSnapshot()
    })

    test('renders heading', () => {
      const { getByRole } = testRenderer('/editor', Editor)
      const heading = getByRole('heading', { level: 1 })
      expect(heading).toBeInTheDocument()
    })

    test('renders copy', () => {
      const { getByText } = testRenderer('/editor', Editor)
      const copy = getByText('Labore labore', {
        selector: 'p',
        exact: false
      })
      expect(copy).toBeInTheDocument()
    })

    test('renders top nav', () => {
      const { getAllByRole } = testRenderer('/editor', Editor)
      const links = getAllByRole('link')
      expect(links.length).toBe(3)
    })

    test('renders file loader heading', () => {
      const { getByText } = testRenderer('/editor', Editor)
      const heading = getByText('File Loader')
      expect(heading).toBeInTheDocument()
    })
  })
})
