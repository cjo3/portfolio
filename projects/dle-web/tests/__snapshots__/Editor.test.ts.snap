// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor render matches current snapshop 1`] = `
<div
  className="MuiGrid-root MuiGrid-container"
>
  <nav
    className="makeStyles-nav-5"
  >
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-navItem-6"
      href="/"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
    >
      <span
        className="MuiButton-label"
      >
        <span
          className="MuiButton-startIcon MuiButton-iconSizeMedium"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            size="small"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
            />
          </svg>
        </span>
        Home
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </a>
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-navItem-6"
      href="/editor"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
    >
      <span
        className="MuiButton-label"
      >
        <span
          className="MuiButton-startIcon MuiButton-iconSizeMedium"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            size="small"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.95 10.78c.03-.25.05-.51.05-.78s-.02-.53-.06-.78l1.69-1.32c.15-.12.19-.34.1-.51l-1.6-2.77c-.1-.18-.31-.24-.49-.18l-1.99.8c-.42-.32-.86-.58-1.35-.78L12 2.34c-.03-.2-.2-.34-.4-.34H8.4c-.2 0-.36.14-.39.34l-.3 2.12c-.49.2-.94.47-1.35.78l-1.99-.8c-.18-.07-.39 0-.49.18l-1.6 2.77c-.1.18-.06.39.1.51l1.69 1.32c-.04.25-.07.52-.07.78s.02.53.06.78L2.37 12.1c-.15.12-.19.34-.1.51l1.6 2.77c.1.18.31.24.49.18l1.99-.8c.42.32.86.58 1.35.78l.3 2.12c.04.2.2.34.4.34h3.2c.2 0 .37-.14.39-.34l.3-2.12c.49-.2.94-.47 1.35-.78l1.99.8c.18.07.39 0 .49-.18l1.6-2.77c.1-.18.06-.39-.1-.51l-1.67-1.32zM10 13c-1.65 0-3-1.35-3-3s1.35-3 3-3 3 1.35 3 3-1.35 3-3 3z"
              transform="scale(1.2, 1.2)"
            />
          </svg>
        </span>
        Editor
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </a>
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-navItem-6"
      href="/error"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
    >
      <span
        className="MuiButton-label"
      >
        <span
          className="MuiButton-startIcon MuiButton-iconSizeMedium"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            size="small"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
            />
          </svg>
        </span>
        Error
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </a>
  </nav>
  <header
    className="MuiGrid-root makeStyles-Editor_header-2 MuiGrid-container MuiGrid-justify-xs-center"
  >
    <h1
      className="MuiTypography-root MuiTypography-h1 MuiTypography-alignCenter"
    >
      Editor
    </h1>
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter"
    >
      Labore labore ut est sit sanctus stet et. Eos lorem ipsum consetetur magna est voluptua. Lorem justo dolor eirmod est est aliquyam. Dolor sit kasd nonumy lorem tempor dolor no duo et, invidunt gubergren rebum at rebum, et nonumy lorem lorem diam ipsum, sadipscing diam voluptua accusam sit sit, dolor.
    </p>
  </header>
  <section
    className="MuiGrid-root makeStyles-section-7 makeStyles-noWorkbookBg-9 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
    data-testid="FileLoader"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-10 MuiGrid-grid-md-8 MuiGrid-grid-lg-6 MuiGrid-grid-xl-4"
    >
      <form
        className="makeStyles-form-10"
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <div
          className="makeStyles-title-11"
        >
          <h3
            className="MuiTypography-root MuiTypography-h3"
          >
            File Loader
          </h3>
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Select a file using the button below
            . The selected file name will be displayed once selected.
          </p>
        </div>
        <div
          className="makeStyles-fileSelect-12"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <label
                htmlFor="file-upload-input"
              >
                Select File
              </label>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
          <p
            className="MuiTypography-root makeStyles-fileName-16 MuiTypography-body1"
          >
            No File Selected
          </p>
          <input
            className="makeStyles-hiddenInput-15"
            id="file-upload-input"
            name="file-upload-input"
            onChange={[Function]}
            type="file"
          />
        </div>
        <div
          className="makeStyles-dropLoader-13"
        >
          <div
            className="makeStyles-dropContainer-20"
            data-testid="DropSelector"
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDrop={[Function]}
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root makeStyles-dropIcon-22"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 9h-4V3H9v6H5l7 7 7-7zM5 18v2h14v-2H5z"
              />
            </svg>
          </div>
        </div>
        <div
          className="makeStyles-actionButtons-14"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-formButton-17 makeStyles-submitButton-18 MuiButton-containedPrimary Mui-disabled Mui-disabled"
            disabled={true}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            type="submit"
          >
            <span
              className="MuiButton-label"
            >
              Load
            </span>
          </button>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-formButton-17 makeStyles-resetButton-19 Mui-disabled Mui-disabled"
            disabled={true}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            type="reset"
          >
            <span
              className="MuiButton-label"
            >
              Reset
            </span>
          </button>
        </div>
      </form>
    </div>
  </section>
  <section
    className="MuiGrid-root makeStyles-section-23 MuiGrid-container MuiGrid-justify-xs-center"
  >
    <h3
      className="MuiTypography-root MuiTypography-h3"
    >
      No file loaded
    </h3>
  </section>
</div>
`;
